util.js

```diff
@@ -34,21 +27,19 @@ renderDOM = (obj, target) => {
     if (typeof obj === "string") {
         dom = new Text(obj);
     } else if (Array.isArray(obj)) {
-        if (obj[0] === "!comment") {
-            dom = new Comment(obj[1]);
-        } else if (obj[0] === "!fragment") {
-            obj.slice(2).forEach((child) => renderDOM(child, target));
+        const [type, attrs, ...rest] = obj;
+        if (type === "!comment") {
+            dom = new Comment(attrs);
+        } else if (type === "!fragment") {
+            rest.forEach((child) => renderDOM(child, target));
         } else {
-            dom = document.createElement(obj[0]);
-            const attrs = obj[1];
-            Object.keys(attrs).forEach(
-                (key) => dom.setAttribute(key, attrs[key]));
-            obj.slice(2).forEach((child) => renderDOM(child, dom));
+            dom = Object.assign(document.createElement(type), attrs);
```

Comment:
this doesn't work like the original, setting a DOM attribute is not the same as assigning properties.
the trivial examples are `"data-x"` and `class`/`className`